imap/http_dblookup.c

```diff
@@ -360,6 +360,106 @@ static int get_uid2emails(struct transaction_t *txn,
     return ret;
 }
 
+static int get_uid2members(struct transaction_t *txn,
+                           const char *userid, const char *uid)
+{
+    struct auth_state *authstate = auth_newstate(userid);
+    int ret = HTTP_NO_CONTENT;
+
+    if (!authstate) goto done;
+
+    struct carddav_db *carddavdb = carddav_open_userid(userid);
+    if (!carddavdb) goto done;
+
+    ptrarray_t *abook_sets = jmap_get_accessible_addressbooks(userid, authstate,
+                                                              &httpd_namespace,
+                                                              carddavdb);
+
+    /* Get members in the given card */
+    strarray_t *member_uids = NULL;
+    struct abook_set *set;
+    mbentry_t *mbentry;
+    const char *member;
+    int found = 0;
+    int i, j, k;
```

Comment:
Some (all?) of these declarations look misplaced in scope. They only are used within the block scopes of the `for` loops, so why declare them here rather than within the blocks where they are required?
The `i,j,k` indexes all could be declared and initialized within the `for` loop statements, too.

Reply:
Fixed